cache.go

```diff
@@ -0,0 +1,19 @@
+package main
+
+type Cache interface {
+	read()
+	write()
+	new()
```

Comment:
Can you please add comments about these functions and it's usage. Also are you sure that we don't require any arguments for it ? As per my understanding `writer()` must be passed some data to writer in the cache right ?